set_env.sh

```diff
@@ -95,15 +116,15 @@ fi
 case $SIMPLE_PLATFORM in
 
     linux)
-        PATH=/usr/local/gcc103/bin:/usr/bin:/bin:/usr/local/gcc103/sbin:/usr/sbin:/sbin:/usr/local/bin
+        PATH=/usr/local/gcc103/bin:/usr/bin:/bin:/usr/local/gcc103/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/cmake/bin
         ;;
 
     macos)
         PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin
         ;;
 
     windows)
-        PATH=/mingw64/bin:/usr/bin:/bin:$ROOT/Windows/System32:$ROOT/Windows
+        PATH=/mingw64/bin:/usr/bin:/bin:$ROOT/Windows/System32:$ROOT/Windows:$ROOT/PROGRA~2/MICROS~2/2019/PROFES~1/Common7/IDE/COMMON~1/MICROS~1/CMake/CMake/bin
```

Comment:
If CMake is installed according to the [build setup instructions](https://github.com/elastic/ml-cpp/blob/main/build-setup/windows.md#cmake) then it ends up being version 3.19 and goes in `C:\Program Files\CMake\bin`:

<img width="760" alt="Screenshot 2022-05-16 at 14 06 39" src="https://user-images.githubusercontent.com/7405510/168599706-3e24d5bc-1794-4c02-9118-6a1d09579b69.png">

If CMake is installed using Visual Studio then it goes in the directory you've specified here, and currently is version 3.20:

<img width="774" alt="Screenshot 2022-05-16 at 14 05 56" src="https://user-images.githubusercontent.com/7405510/168599881-02a7a840-f25c-4ecf-84d8-8117b3314c4c.png">

We _could_ change the build instructions to tell people to install CMake with Visual Studio.  That might affect the scripts that build the CI workers as we might need to specify an extra option when doing the unattended Visual Studio install.  If we go this way we should also change the build setup instructions to tell people to always use that CMake on Windows.

However, on the other hand we may benefit in the future from having CMake decoupled from Visual Studio.  If we install our own then we can potentially upgrade CMake without having to upgrade Visual Studio.  I think having CMake decoupled is better _unless_ Microsoft has fixed something in their build of CMake that makes it work better with Visual Studio than vanilla CMake.  In that case we _should_ use Microsoft's build of CMake, accepting that Microsoft now dictates our CMake upgrade policy and should have comments to say that wherever we tell people to install CMake on other platforms, e.g. "On Windows we are constrained to use CMake 3.20, so no platform can use a CMake feature that was introduced after this."

Reply:
Agreed, I think unless there's a very good reason not to then we should use our own build of CMake on all platforms.